### PR TITLE
Ms.plot load perf2

### PR DESCRIPTION
**Pull request by mariano54**
Alternate PR to https://github.com/Chia-Network/chia-blockchain/pull/10847

### CODE REVIEW

File: chia/plotting/create_plots.py [hunk 1]
```diff
@@ -141,20 +142,46 @@ async def resolve_plot_keys(
...
+                log.info(f"NOT adding directory {resolved_final_dir} to harvester for farming")
+                if resolved_final_dir in plot_directories_list:
+                    log.warning(
+                        f"Directory {resolved_final_dir} already exists for harvester, please remove it manually"
+                    )
+            else:
+                if resolved_final_dir not in plot_directories_list:
+                    # Adds the directory to the plot directories if it is not present
+                    log.info(f"Adding directory {resolved_final_dir} to harvester for farming")
+                    add_plot_directory(root_path, resolved_final_dir)
```
Comment (mariano54) [inline chia/plotting/create_plots.py:169]: Actually this will lock the config also so we might run into an issue

Comment (mariano54) [reply chia/plotting/create_plots.py]: Fixed now.

File: tests/block_tools.py [hunk 2]
```diff
@@ -326,14 +328,14 @@ async def new_plot(
             assert plot_id_new is not None
             assert path_new is not None
 
+            if not exclude_plots:
```
Comment (mariano54) [inline tests/block_tools.py:331]: this is wrong, i guess bad merge

File: tests/block_tools.py [hunk 3]
```diff
@@ -307,14 +310,13 @@ async def new_plot(
...
                 plot_keys,
-                self.root_path,
                 use_datetime=False,
                 test_private_keys=[AugSchemeMPL.key_gen(std_hash(self.created_plots.to_bytes(2, "big")))],
             )
             self.created_plots += 1
 
             plot_id_new: Optional[bytes32] = None
-            path_new: Optional[Path] = None
+            path_new: Path = Path()
```
Comment (xdustinface) [inline tests/block_tools.py:319]:
```suggestion
            path_new: Optional[Path] = None
```

My bad, not sure how i ended up introducing this but its not related here.

File: tests/block_tools.py [hunk 4]
```diff
@@ -256,7 +261,6 @@ async def setup_plots(self):
             await self.new_plot(
                 path=self.plot_dir / "not_in_keychain",
                 plot_keys=PlotKeys(G1Element(), G1Element(), None),
```
Comment (xdustinface) [inline tests/block_tools.py:263]:
```suggestion
                plot_keys=PlotKeys(G1Element(), G1Element(), None),
                exclude_plots=True,
```

This fixes the tests.

**Review by xdustinface — APPROVED** [review]

**Pull request merged**